### PR TITLE
Use "Redmean" color distance calculation for GetPALColorId()

### PR DESCRIPTION
**Pull request by Districh-ru**
Relates to https://github.com/ihhub/fheroes2/issues/7574

This PR modifies the color distance calculation formula to better fit human perception (https://en.wikipedia.org/wiki/Color_difference).

One disadvantage of this PR is that it takes about 40% more time to calculate `rgbToId` array data (120 ms vs 85 ms) during game load.

For the comparison:
Ideal color transitions (click to view in non-scaled size):
testBest
test2Best

Master build:
testOrig
test2Orig

This PR:
testRedmean
test2Redmean

The change is noticeable for the dark colors: (there are less noticeable flashes wile fading the credits page)

Master build:
fheroes2 master 2023-08-23 14-33-36-140

This PR:
fheroes2 PR 2023-08-23 14-31-45-500

<details><summary>Another comparison (castle screen)</summary>

Master build:
fheroes2 master 2023-08-23 14-09-19-740

This PR:
fheroes2 PR 2023-08-23 14-10-36-540

</details>

### CODE REVIEW

File: src/engine/image.cpp [hunk 1]
```diff
@@ -352,31 +352,35 @@ namespace
...
                 uint32_t bestPos = 0;
 
+                // Use the "No cycle" palette.
                 const uint8_t * correctorX = transformTable + 256 * 15;
 
                 for ( uint32_t i = 0; i < 256; ++i, ++correctorX ) {
                     const uint8_t * palette = gamePalette + static_cast<ptrdiff_t>( *correctorX ) * 3;
 
-                    const int32_t offsetRed = static_cast<int32_t>( *palette ) - static_cast<int32_t>( r );
+                    const int32_t sumRed = ( static_cast<int32_t>( *palette ) + r );
```
Comment (ihhub) [inline src/engine/image.cpp:374]: Please remove parenthesis for this line or add it for all other cases.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @Districh-ru , I put just one minor remark here. Could you please check it?

File: src/engine/image.cpp (hunk 1 above)
Comment (Districh-ru) [reply src/engine/image.cpp]: Hi, @ihhub, sure)

**Review by ihhub — APPROVED** [review]

**Review by idshibanov — APPROVED** [review]

**Comment by ihhub** [comment]
@Districh-ru , thank you so much for this improvement!

**Pull request merged**